lib/graphql/analysis/ast/field_usage.rb

```diff
@@ -7,21 +7,45 @@ def initialize(query)
           super
           @used_fields = Set.new
           @used_deprecated_fields = Set.new
+          @used_deprecated_arguments = Set.new
         end
 
         def on_leave_field(node, parent, visitor)
           field_defn = visitor.field_definition
           field = "#{visitor.parent_type_definition.graphql_name}.#{field_defn.graphql_name}"
           @used_fields << field
           @used_deprecated_fields << field if field_defn.deprecation_reason
+
+          extract_deprecated_arguments(visitor.query.arguments_for(node, visitor.field_definition).argument_values)
         end
 
         def result
           {
             used_fields: @used_fields.to_a,
-            used_deprecated_fields: @used_deprecated_fields.to_a
+            used_deprecated_fields: @used_deprecated_fields.to_a,
+            used_deprecated_arguments: @used_deprecated_arguments.to_a,
           }
         end
+
+        private
+
+        def extract_deprecated_arguments(argument_values)
+          argument_values.each_pair do |_argument_name, argument|
+            if argument.definition.deprecation_reason
+              field = "#{argument.definition.owner.graphql_name}##{argument.definition.graphql_name}"
+              @used_deprecated_arguments << field
```

Comment:
Yes, best to use `.` -- that's what we use other places, for example, `GraphQL::Schema::Argument#path` outputs something like `ObjectType.fieldName.argumentName`.  Honestly, you could probably use _that_:

```suggestion
              @used_deprecated_arguments << argument.definition.path
```

(`.path` was added after this analyzer was written, so if other code here doesn't use that method, that's probably why.)

Reply:
Fixed! [`9cc20ce` (#3549)](https://github.com/rmosolgo/graphql-ruby/pull/3549/commits/9cc20cebf9448fe50a655719d48ba566d8d40879)